akd/src/storage/types.rs

```diff
@@ -85,11 +82,8 @@ impl crate::storage::Storable for ValueState {
         }
         let epoch_bytes: [u8; 8] = bin[1..=8].try_into().expect("Slice with incorrect length");
         let epoch = u64::from_be_bytes(epoch_bytes);
-        if let Ok(username) = std::str::from_utf8(&bin[9..]) {
-            Ok(ValueStateKey(username.to_string(), epoch))
-        } else {
-            Err("Invalid string format".to_string())
-        }
+        let username = Vec::from(&bin[9..]);
+        Ok(ValueStateKey(username, epoch))
```

Comment:
Notable here, both before and after, is that there's no guard that the parameter `bin` is long enough.

Reply:
Will an empty slice still be valid? If not we should add a guard to protect against this (there must be SOMEthing for a user label)

Reply:
Oh, huh. I lost the forest for the trees at some point, there's literally already a guard above on (new) line 80. 🥸